src/test/java/htsjdk/samtools/CRAMContainerStreamWriterTest.java

```diff
@@ -91,6 +91,28 @@ private void doTest(final List<SAMRecord> samRecords, final ByteArrayOutputStrea
         Assert.assertEquals(count, samRecords.size());
     }
 
+    private void doTest(final List<SAMRecord> samRecords, final ByteArrayOutputStream outStream, final SAMFileHeader header, final AbstractCRAMIndexer indexer) {
+        final ReferenceSource refSource = createReferenceSource();
+
+        final CRAMContainerStreamWriter containerStream = new CRAMContainerStreamWriter(outStream, refSource, header, "test", indexer);
+        containerStream.writeHeader(header);
```

Comment:
From this line down to the end of the method is duplicated in both `doTest` overloads, and can be extracted into a common method.